pennylane/devices/qutrit_mixed/sampling.py

```diff
@@ -0,0 +1,334 @@
+    state_wires = qml.wires.Wires(range(total_indices))
+
+    wires_to_sample = wires or state_wires
+    num_wires = len(wires_to_sample)
+    basis_states = np.arange(QUDIT_DIM**num_wires)
+
+    with qml.queuing.QueuingManager.stop_recording():
+        probs = measure(qml.probs(wires=wires_to_sample), state, is_state_batched)
+
+    if is_state_batched:
+        # Produce separate keys for each of the probabilities along the broadcasted axis
+        keys = []
+        for _ in state:
+            key, subkey = jax.random.split(key)
+            keys.append(subkey)
+        samples = jnp.array(
+            [
+                jax.random.choice(_key, basis_states, shape=(shots,), p=prob)
+                for _key, prob in zip(keys, probs)
+            ]
+        )
+    else:
+        samples = jax.random.choice(key, basis_states, shape=(shots,), p=probs)
+
+    res = np.zeros(samples.shape + (num_wires,), dtype=np.int64)
+    for i in range(num_wires):
+        res[..., -(i + 1)] = (samples // (QUDIT_DIM**i)) % QUDIT_DIM
+    return res
+
+
+def sample_state(
+    state,
+    shots: int,
+    is_state_batched: bool = False,
+    wires=None,
+    rng=None,
+    prng_key=None,
+) -> np.ndarray:
+    """Returns a series of samples of a state.
+
+    Args:
+        state (array[complex]): A state vector to be sampled
+        shots (int): The number of samples to take
+        is_state_batched (bool): whether the state is batched or not
+        wires (Sequence[int]): The wires to sample
+        rng (Union[None, int, array_like[int], SeedSequence, BitGenerator, Generator]):
+            A seed-like parameter matching that of ``seed`` for ``numpy.random.default_rng``.
+            If no value is provided, a default RNG will be used
+        prng_key (Optional[jax.random.PRNGKey]): An optional ``jax.random.PRNGKey``. This is
+            the key to the JAX pseudo random number generator. Only for simulation using JAX.
+
+    Returns:
+        ndarray[int]: Sample values of the shape (shots, num_wires)
+    """
+    if prng_key is not None:
+        return _sample_state_jax(
+            state, shots, prng_key, is_state_batched=is_state_batched, wires=wires
+        )
+
+    rng = np.random.default_rng(rng)
+
+    total_indices = get_num_wires(state, is_state_batched)
+    state_wires = qml.wires.Wires(range(total_indices))
+
+    wires_to_sample = wires or state_wires
+    num_wires = len(wires_to_sample)
+    basis_states = np.arange(QUDIT_DIM**num_wires)
+
+    with qml.queuing.QueuingManager.stop_recording():
+        probs = measure(qml.probs(wires=wires_to_sample), state, is_state_batched)
+
+    if is_state_batched:
+        # rng.choice doesn't support broadcasting
+        samples = np.stack([rng.choice(basis_states, shots, p=p) for p in probs])
+    else:
+        samples = rng.choice(basis_states, shots, p=probs)
+
+    res = np.zeros(samples.shape + (num_wires,), dtype=np.int64)
```

Comment:
This one also looks the same as the qubit version (minus the small differences needed for making it work with qudits)?

Reply:
Yes, I worked off of that as the logic would be similar, it would keep things more consistent and it saves time in development.